pkg/playground/react-demo-typeahead.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+
+import React, { useState } from "react";
+import { createRoot, Container } from 'react-dom/client';
+
+import { Checkbox } from '@patternfly/react-core';
+import { TypeaheadSelect, TypeaheadSelectOption } from "cockpit-components-typeahead-select";
+
+const TypeaheadDemo = ({ options } : { options: TypeaheadSelectOption[] }) => {
```

Comment:
I'm really amazed that you could fully type this without too much effort. Makes the "type ahead demo" twice as interesting! :wink: